docs/5.1. Transport - MQTT.md

```diff
@@ -0,0 +1,77 @@
+
+This document describes the use of the MQTT transport.
+
+Other sections can be accessed from the [Overview](1.0.%20Overview.md).
+
+## 1. Introduction
+
+* Simple and lightweight
+* Broker based
+* Can be used on top of TCP or WebSockets
+* Can use encryption
+* Supports various QOS settings for message delivery
+* Inherently solves the problem of late joiners using retained messages
+* Events filtering is handled by the broker with topic filtering mechanics
+
+## 2. NMOS Resources transport
+
+NMOS resources using MQTT will use the `urn:x-nmos:transport:mqtt` transport.
+
+## 3. Connection management
+
+Only IS-05 connection management will be used for connection management of resources defined in this specification.
+
+### 3.1 broker_topic
+
+The `broker_topic` parameter will hold the MQTT topic and will always be set to the source id, prefixed with `x-nmos/source/` for easier filtering.
+
+### 3.2 rest_api_url
+
+The `rest_api_url` parameter will represent the url to the API path which offers the current state and type of an event emitter (source) (see [Event & Tally REST API](6.0.%20Event_and_tally_rest_api.md))
+
+It is important for the controller to always populate the `rest_api_url` field using the following template:  
+`{is_05_control_base_url}`sources/`{source_id_associated_with_sender}`/  
+The controller will populate `is_05_control_base_url` as being the href offered by the `urn:x-nmos:events:query/v1.0` control in the controls array of the sender device.  
+The controller will populate `source_id_associated_with_sender` as being the unique ID of the source associated with the sender.  
+For consistency the `rest_api_url` url offered will always end with a trailing slash.
+
+A receiver should always expect the `rest_api_url` to follow the format above and only needs to append one of the following suffixes:
+
+* state (to retrieve the current state of the emitter)
+* type (to retrieve the metadata associated with the event_type of the emitter)
+
+_Example of IS-05 PATCH request_
+
+```json
+{
+    "sender_id": "9f463872-9621-4939-aa3a-dc3c82d8578b",
+    "master_enable": true,
+    "activation": {
+        "mode": "activate_immediate",
+        "requested_time": null
+    },
+    "transport_params": [
+        {
+            "broker_topic": "x-nmos/source/9f463872-9621-4939-aa3a-dc3c82d8578b",
+            "rest_api_url": "http://hostname/x-nmos/events/v1.0/sources/9f463872-9621-4939-aa3a-dc3c82d8578b/"
+        }
+    ]
+}
+```
+
+## 4. QOS Settings
+
+MQTT publishers are recommended to use the `exactly once QOS (2)`
+
+## 5. Late joiners
+
+MQTT publishers are required to send the `retained message` flag with every message in order to solve the problem of late joiners.
+
+Consumers (receivers) have a choice to either use the retained message or query the late joiners api using `rest_api_url` to access the latest state of an emitter in order to get in sync.
```

Comment:
Maybe we should mention the will message here and add a reference to the messages document?

Reply:
Made the necessary changes as per conversations via email (github comments were not working)